sage.py

```diff
@@ -203,9 +203,15 @@ def group_alerts_per_team(alerts, port_mapping):
             else:
                 dst_port = port_mapping[dst_port]['name']
 
-            # TODO: add the check for 10.0.254 in src_ip or in dst_ip - if not, then discard
-            # TODO: If present in src_ip, then add (src_ip, dst_ip). If in dst_ip, then add (dst_ip, src_ip)
+            # For CPTC dataset, attacker IPs (src_ip) start with '10.0.254', but this prefix might be in dst_ip
             # TODO: for the future, we might want to address internal paths
+            if dataset_name == 'cptc' and not src_ip.startswith('10.0.254') and not dst_ip.startswith('10.0.254'):
+                continue
+            if dataset_name == 'cptc' and dst_ip.startswith('10.0.254'):
+                temp = dst_ip
+                dst_ip = src_ip
+                src_ip = temp
+
             if (src_ip, dst_ip) not in host_alerts.keys() and (dst_ip, src_ip) not in host_alerts.keys():
```

Comment:
Okay, here I need some help. From this code fragment also performs swapping. I am not sure if it is needed now that the swapping of `src_ip` and `dst_ip` happens in this function. Shouldn't it be this way (although the episode traces are affected):

![image](https://github.com/tudelft-cda-lab/SAGE/assets/45541174/e7ca8d88-f5df-4ee6-aae6-ccb12084580e)

![image](https://github.com/tudelft-cda-lab/SAGE/assets/45541174/834713a4-904d-4856-b279-2d8dad15136d)

Reply:
For CPTC-2017, for example, there is a graph that is affected even if we remove IDs.

With the current version:
![updated-2017 txt-attack-graph-for-victim-10 0 254 21-DATAEXFILTRATIONmicrosoftds](https://github.com/tudelft-cda-lab/SAGE/assets/45541174/3ba487d0-66d6-4aeb-832a-abcbaf7c9258)

With the version I mentioned in the comment:
![updated-2-2017 txt-attack-graph-for-victim-10 0 254 21-DATAEXFILTRATIONmicrosoftds](https://github.com/tudelft-cda-lab/SAGE/assets/45541174/e6a26ebc-d4b2-4646-a79c-5692e2ef4f69)

Reply:
For CCDC, some graphs are also affected by this change, even if IDs are removed:
![image](https://github.com/tudelft-cda-lab/SAGE/assets/45541174/4523df11-4fbe-4b68-b173-1a4d364fef54)

Reply:
TODO: if (a, b) exists in the list, then (b, a) cannot exist

Reply:
The check will stay, as discussed in the meeting